src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/CollectionFloatingBarplotPlugin.java

```diff
@@ -0,0 +1,115 @@
+import org.veupathdb.service.eda.ds.Resources;
+import org.veupathdb.service.eda.ds.plugin.AbstractEmptyComputePlugin;
+import org.veupathdb.service.eda.ds.plugin.AbstractPlugin;
+import org.veupathdb.service.eda.ds.plugin.standalonemap.markers.OverlaySpecification;
+import org.veupathdb.service.eda.generated.model.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+import static org.veupathdb.service.eda.common.plugin.util.RServeClient.streamResult;
+import static org.veupathdb.service.eda.common.plugin.util.RServeClient.useRConnectionWithRemoteFiles;
+import static org.veupathdb.service.eda.ds.metadata.AppsMetadata.VECTORBASE_PROJECT;
+
+public class FloatingBarplotPlugin extends AbstractEmptyComputePlugin<FloatingBarplotPostRequest, FloatingBarplotSpec> {
+
+  @Override
+  public String getDisplayName() {
+    return "Bar plot";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Visualize the distribution of a group of categorical variables";
+  }
+
+  @Override
+  public List<String> getProjects() {
+    return List.of(VECTORBASE_PROJECT);
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
+    return new ConstraintSpec()
+      .dependencyOrder(List.of("xAxisVariable"), List.of("overlayVariable"))
+      .pattern()
+        .element("xAxisVariable")
+          .maxValues(10)
+          .description("Variable Group vocabulary must have 10 or fewer unique values.")
+        .element("overlayVariable")
+      .done();
+  }
+
+  @Override
+  protected AbstractPlugin<FloatingBarplotPostRequest, FloatingBarplotSpec, Void>.ClassGroup getTypeParameterClasses() {
+    return new ClassGroup(FloatingBarplotPostRequest.class, FloatingBarplotSpec.class, Void.class);
+  }
+
+  @Override
+  protected void validateVisualizationSpec(FloatingBarplotSpec pluginSpec) throws ValidationException {
+    validateInputs(new DataElementSet()
+      .entity(pluginSpec.getOutputEntityId())
+      .var("overlayVariable", Optional.ofNullable(pluginSpec.getCollectionOverlayConfig())
+          .map(CollectionOverlayConfig::getCollection())
+          .orElse(null)));
+    if (pluginSpec.getCollectionOverlayConfig() != null) {
+      // TODO replace this w collection specific validation. maybe leave a stub for now, and reuse some collection marker validation once merged.
+    }
+    if (pluginSpec.getBarMode() == null) {
+      throw new ValidationException("Property 'barMode' is required.");
+    }
+  }
+
+  @Override
+  protected List<StreamSpec> getRequestedStreams(FloatingBarplotSpec pluginSpec) {    
+    return ListBuilder.asList(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
+        .addVars(getUtil().getChildrenVariables(pluginSpec.getCollectionOverlayConfigWithValues().getCollection())
```

Comment:
when do we filter the data based on the chosen values? Has that already happened?

Reply:
In R I guess...